sentence_transformers/SentenceTransformer.py

```diff
@@ -253,6 +254,7 @@ def encode(
         batch_size: int = 32,
         show_progress_bar: bool = None,
         output_value: str = "sentence_embedding",
+        precision: Literal["float32", "int8", "uint8", "binary", "ubinary"] = "float32",
```

Comment:
Yes :100:% aligned on this, otherwise it will break for users passing positional args. In `huggingface_hub` we have an helper [`_deprecate_positional_args`](https://github.com/huggingface/huggingface_hub/blob/6d69291c77d41ac1b9bf1f1c07e8c365fc5e1e5d/src/huggingface_hub/utils/_deprecation.py#L7) to easily deprecate such a behavior, while preserving a backward compatibility. We usually do that for a few versions and then remove the possibility completely.

Reply:
The new `precision` argument will only break once someone has provided ~6 other positional arguments. That said, in the long term I am certainly in favour of deprecating positional arguments altogether for `encode`, e.g. by adding `*` as the second option. For this PR, I'd like to keep it this way. Does that seem okay @osanseviero @Wauplin? Otherwise I can move it to the very end, but I believe it's more intuitive in the middle here.

Reply:
> The new precision argument will only break once someone has provided ~6 other positional arguments.

Yep that's the case. Hopefully no-one does that but you can't tell for sure. The `_deprecate_positional_args` suggested above allows for a transition period where both positional and keyword arguments are accepted. If positional args are passed, a deprecation warning is raised but the script executes correctly. You would still need to put `precision` last while the deprecation period lasts.

Reply:
I'm a bit wary to already pick a full deprecation version, as I'm not sure how fast releases will happen, so that's why I haven't applied `_deprecate_positional_args` yet.

Reply:
Fair enough!